src/nunavut/lang/cpp/templates/_fields_as_union_specializations.j2

```diff
@@ -0,0 +1,60 @@
+{#-
+ # Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ # Copyright (C) 2021  UAVCAN Development Team  <uavcan.org>
+ # This software is distributed under the terms of the MIT License.
+#}
+// +-------------------------------------------------------------------------------------------------------------------+
+// | EXPLICIT TEMPLATE SPECIALIZATION
+// +-------------------------------------------------------------------------------------------------------------------+
+{%- for field in composite_type.fields_except_padding %}
+template<>
+constexpr typename {{ composite_type | short_reference_name }}::VariantType::alternative<{{ loop.index0 }}>::type* {{ composite_type | short_reference_name }}::VariantType::get_if<
+        {{ composite_type | short_reference_name }}::VariantType::IndexOf::{{ field.name | id }}
+    >({{ composite_type | short_reference_name }}::VariantType* v) noexcept
+{
+    {{ field.data_type | declaration }}* result = nullptr;
+    if (v)
+    {
+        {{ composite_type | short_reference_name }}::VariantType& vref = *v;
+        if(vref.tag_ == {{ composite_type | short_reference_name }}::VariantType::IndexOf::{{ field.name | id }})
+        {
+            static_assert(std::is_same<typename {{ composite_type | short_reference_name }}::VariantType::alternative<{{ loop.index0 }}>::type, {{ field.data_type | declaration }}>::value,
+                          "The alternative type determined by the preprocessor disagrees with the generated type.");
```

Comment:
I think you meant "compiler" instead of "preprocessor"

Reply:
fixed